kidviz/views.py

```diff
@@ -1100,3 +1100,44 @@ class StartNewObservation(LoginRequiredMixin, View):
     def get(self, request):
         self.request.session['create_new'] = True
         return HttpResponseRedirect(reverse_lazy('observation_view'))
+
+
+class SaveUserSetupView(LoginRequiredMixin, FormView):
+    form_class = SetupSaveForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST, instance=self.get_setup_instance())
+
+        if form.is_valid():
+            return self.form_valid(form)
```

Comment:
`FormView` can do it for you. You can just override `form_valid` and `form_invalid` if you need. Also, you can override `get_object` (I'm not sure if exists), `get_form` or `get_form_kwargs`.